src/wallet3/keyring.cpp

```diff
@@ -394,6 +391,58 @@ namespace wallet
       throw std::runtime_error("RCT signing went wrong -- verRctNonSemanticsSimple returned false");
   }
 
+  std::vector<crypto::public_key> Keyring::get_subaddress_spend_public_keys(uint32_t account, uint32_t begin, uint32_t end) {
+    if (begin <= end)
```

Comment:
this comparison seems backwards.  also, make sure you're consistent with inclusive versus exclusive ranges, e.g. does (0,3) mean {0,1,2,3} or {0,1,2}.  the `reserve` call and the loop below may need updated based on that decision.